File: web/src/services/salonOwnershipService.js
import api from './api';

const salonOwnershipService = {
  /**
   * Get all salons owned by current user
   */
  getMySalons: async () => {
    const response = await api.get('/my-salons');
    return response;
  },

  /**
   * Get single salon details
   */
  getMySalon: async (id) => {
    const response = await api.get(`/my-salons/${id}`);
    return response;
  },

  /**
   * Set primary salon
   */
  setPrimarySalon: async (id) => {
    const response = await api.put(`/my-salons/${id}/set-primary`);
    return response;
  },

  /**
   * Add a new salon
   */
  addSalon: async (salonData) => {
    const response = await api.post('/my-salons', salonData);
    return response;
  },

  /**
   * Check if user has access to salon
   */
  checkSalonAccess: async (id) => {
    const response = await api.get(`/my-salons/${id}/check-access`);
    return response;
  },

  /**
   * Transfer salon ownership
   */
  transferOwnership: async (id, newOwnerEmail) => {
    const response = await api.post(`/my-salons/${id}/transfer`, {
      newOwnerEmail,
    });
    return response;
  },

  /**
   * Grant access to another user
   */
  grantAccess: async (id, userEmail, role = 'manager', permissions = {}) => {
    const response = await api.post(`/my-salons/${id}/grant-access`, {
      userEmail,
      role,
      permissions,
    });
    return response;
  },

  /**
   * Revoke user access
   */
  revokeAccess: async (id, userId) => {
    const response = await api.delete(`/my-salons/${id}/revoke-access/${userId}`);
    return response;
  },
};

export default salonOwnershipService;

File: mobile/lib/services/salon_ownership_service.dart
import 'api_service.dart';

class SalonOwnershipService {
  final _api = ApiService();

  Future<Map<String, dynamic>> getMySalons() async {
    try {
      final response = await _api.get('/my-salons');
      if (response['success'] == true) {
        return {
          'success': true,
          'data': response['data']['salons'] ?? [],
        };
      }
      return {'success': false, 'data': []};
    } catch (e) {
      return {'success': false, 'data': [], 'error': e.toString()};
    }
  }

  Future<Map<String, dynamic>?> getMySalon(String id) async {
    try {
      final response = await _api.get('/my-salons/$id');
      if (response['success'] == true) {
        return response['data'];
      }
      return null;
    } catch (e) {
      return null;
    }
  }

  Future<bool> setPrimarySalon(String id) async {
    try {
      final response = await _api.put('/my-salons/$id/set-primary');
      return response['success'] == true;
    } catch (e) {
      return false;
    }
  }

  Future<Map<String, dynamic>> addSalon(Map<String, dynamic> salonData) async {
    try {
      final response = await _api.post('/my-salons', salonData);
      if (response['success'] == true) {
        return {'success': true, 'salon': response['data']['salon']};
      }
      throw Exception(response['message'] ?? 'Failed to add salon');
    } catch (e) {
      return {'success': false, 'error': e.toString()};
    }
  }
}

